### PR TITLE
Remove projects from list package output

### LINKED ISSUE #12585: dotnet list package --format json --include-transitive includes project references as transitive packages.
### NuGet Product Used

dotnet.exe

### Product Version

7.0.203

### Worked before?

no

### Impact

It's more difficult to complete my work

### Repro Steps & Context

If I have a project that references another project, then the beforementioned command states that referenced project is a transitive package dependency for referencing project.

I'm not sure if this is really bug, perhaps it's just a need of some differentiation between project and package in output, for example, additional "type" property in "transitivePackages" node item, saying if it's a "Project" reference or a "Package".

Steps to reproduce:
1. Download the archive from attachment and unpack it somewhere
2. Run dotnet restore in unpacked folder
3. Run dotnet list package --format json --include-transitive

The output says that "DotnetListJsonTransitivesIssue" project has "ReferencedProject" in its transitive packages (or at least doesn't provide an information if this is really a package from some nuget source or just a project). 

Also, if it should include projects in output, in this very case the "ReferencedProject" should be within the "topLevelPackages" node

[DotnetListJsonTransitivesIssue.zip](https://github.com/NuGet/Home/files/11465827/DotnetListJsonTransitivesIssue.zip)

### Verbose Logs

```shell
{
  "version": 1,
  "parameters": "--include-transitive",
  "projects": [
    {
      "path": "C:/Users/User/source/repos/exp/DotnetListJsonTransitivesIssue/DotnetListJsonTransitivesIssue/DotnetListJsonTransitivesIssue.csproj",
      "frameworks": [
        {
          "framework": "net7.0",
          "transitivePackages": [
            {
              "id": "ReferencedProject",
              "resolvedVersion": "1.0.0"
            }
          ]
        }
      ]
    },
    {
      "path": "C:/Users/User/source/repos/exp/DotnetListJsonTransitivesIssue/ReferencedProject/ReferencedProject.csproj",
      "frameworks": [
        {
          "framework": "net7.0"
        }
      ]
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by Greybird**
## Bug

Fixes: https://github.com/NuGet/Home/issues/12585
Other PR: https://github.com/NuGet/NuGet.Client/pull/5302
Design spec: https://github.com/NuGet/Home/pull/12735

Regression? Last working version:

## Description
`list` command, when used with the `--include-transitive` flag, but without any of the `--deprecated`, `--outdated` or `--vulnerable` ones, reports directly referenced projects as transitive dependencies.
This PR removes this behavior, and adds a test to ensure that with or without `--include-transitive` flag, a project is never returned by the `list` command

## PR Checklist

- [x] PR has a meaningful title
- [X] PR has a linked issue.
- [X] Described changes

- **Tests**
  - [X] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [X] N/A

### CODE REVIEW

**Review by zivkan — DISMISSED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs [hunk 1]
```diff
@@ -652,11 +652,10 @@ internal static bool IsPackageReferenceProject(Project project)
         /// <param name="userInputFrameworks">A list of frameworks</param>
         /// <param name="assetsFile">Assets file for all targets and libraries</param>
         /// <param name="transitive">Include transitive packages/projects in the result</param>
-        /// <param name="includeProjects">Include project references in top-level and transitive package lists</param>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs:655]: So in some places, we seem to want to include projects only in the default `dotnet list package`, but in other places, we seem to filter them out. 

The spec did not clearly call out the whole `dotnet list package` lists projects, but `--outdated`, `--vulnerable` or `--deprecated` does not. 

But maybe there is something to be discussed here?

Comment (zivkan) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs:655]: Sorry, I don't understand why we ever want to include projects in `dotnet list package`. There's already a `dotnet list reference` command to show project references.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs:655]: 🤷 The feature design does not explicitly call out either one. The implementation tries to do both. 

I'm ok with listing just the packages, given the list reference command is available, but it just made me wonder whether anyone might truly depend on this.

**Comment by Greybird** [comment]
I see that the checks are having test failures, but they are passing locally, and I seem to have no way to get access to the results. If an action is needed from me, please ping me!

**Comment by zivkan** [comment]
The "Apex Tests on Windows" are just flakey, ignore those.

For Windows func tests, the test `Dotnet.Integration.Test.DotnetListPackageTests.DotnetListPackage_ProjectReference_Succeeds` failed with error message:

```text
Assert.Contains() Failure
Not found: ProjectB
In value:  Project 'ProjectA' has the following package references
[net46]: No packages were found for this framework.
```

[The test only runs on Windows](https://github.com/NuGet/NuGet.Client/blob/a39baac1e0fc3126a767b7261beb3804a28e4a97/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs#L289-L294), so if you're using a different OS, then running test project will skip that test, although it's not clear to me why this test is Windows only. It looks extremely generic and should work the same on Linux & Mac.

But [the test also looks fundamentally flawed to me](https://github.com/NuGet/NuGet.Client/blob/a39baac1e0fc3126a767b7261beb3804a28e4a97/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs#L325). `git blame` says I wrote the test! I added it when I fixed a bug, but now I believe that it was just an easy way to hit the bug (use `--outdated` on a package that doesn't have any known versions on any package source). Though I was newish on the team at the time, so I'm not sure I would have questioned the weird behaviour at the time. Anyway, IMO the test should be changed to validate that `dotnet list package --show-transitive` does NOT show project references. This will ensure that nobody re-introduces this behaviour accidentally in the future.

**Comment by Greybird** [comment]
@zivkan Thanks for the info!
As I had added a `DotnetListPackage_DoesNotReturnProjects` test to validate no projects were returned, I chose to remove `DotnetListPackage_ProjectReference_Succeeds `, and to add tests cases to mine to further validate the situation for no flags/`--include-transitive` for all (currently supported) report types.

**Review by zivkan — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Review by martinrrm — APPROVED** [review]

**Pull request merged**